In [2]:
import pandas as pd
import matplotlib.pyplot as plt

In [17]:
df_test = pd.read_csv("test.csv")

### Data Exlporation

In [19]:
df_test.head(20)

PassengerId  Pclass                                               Name  \
0           892       3                                   Kelly, Mr. James   
1           893       3                   Wilkes, Mrs. James (Ellen Needs)   
2           894       2                          Myles, Mr. Thomas Francis   
3           895       3                                   Wirz, Mr. Albert   
4           896       3       Hirvonen, Mrs. Alexander (Helga E Lindqvist)   
5           897       3                         Svensson, Mr. Johan Cervin   
6           898       3                               Connolly, Miss. Kate   
7           899       2                       Caldwell, Mr. Albert Francis   
8           900       3          Abrahim, Mrs. Joseph (Sophie Halaut Easu)   
9           901       3                            Davies, Mr. John Samuel   
10          902       3                                   Ilieff, Mr. Ylio   
11          903       1                         Jones, Mr. Charles Cresson   
12          904       1      Snyder, Mrs. John Pillsbury (Nelle Stevenson)   
13          905       2                               Howard, Mr. Benjamin   
14          906       1  Chaffee, Mrs. Herbert Fuller (Carrie Constance...   
15          907       2      del Carlo, Mrs. Sebastiano (Argenia Genovesi)   
16          908       2                                  Keane, Mr. Daniel   
17          909       3                                  Assaf, Mr. Gerios   
18          910       3                       Ilmakangas, Miss. Ida Livija   
19          911       3              Assaf Khalil, Mrs. Mariana (Miriam")"   

       Sex   Age  SibSp  Parch            Ticket     Fare Cabin Embarked  
0     male  34.5      0      0            330911   7.8292   NaN        Q  
1   female  47.0      1      0            363272   7.0000   NaN        S  
2     male  62.0      0      0            240276   9.6875   NaN        Q  
3     male  27.0      0      0            315154   8.6625   NaN        S  
4   female  22.0      1      1           3101298  12.2875   NaN        S  
5     male  14.0      0      0              7538   9.2250   NaN        S  
6   female  30.0      0      0            330972   7.6292   NaN        Q  
7     male  26.0      1      1            248738  29.0000   NaN        S  
8   female  18.0      0      0              2657   7.2292   NaN        C  
9     male  21.0      2      0         A/4 48871  24.1500   NaN        S  
10    male   NaN      0      0            349220   7.8958   NaN        S  
11    male  46.0      0      0               694  26.0000   NaN        S  
12  female  23.0      1      0             21228  82.2667   B45        S  
13    male  63.0      1      0             24065  26.0000   NaN        S  
14  female  47.0      1      0       W.E.P. 5734  61.1750   E31        S  
15  female  24.0      1      0     SC/PARIS 2167  27.7208   NaN        C  
16    male  35.0      0      0            233734  12.3500   NaN        Q  
17    male  21.0      0      0              2692   7.2250   NaN        C  
18  female  27.0      1      0  STON/O2. 3101270   7.9250   NaN        S  
19  female  45.0      0      0              2696   7.2250   NaN        C

In [6]:
df_test["Embarked"].value_counts()

S    270
C    102
Q     46
Name: Embarked, dtype: int64

In [7]:
df_test["Pclass"].value_counts()

3    218
1    107
2     93
Name: Pclass, dtype: int64

In [9]:
df_test.shape

(418, 11)

In [11]:
df_test.isnull().sum()

PassengerId      0
Pclass           0
Name             0
Sex              0
Age             86
SibSp            0
Parch            0
Ticket           0
Fare             1
Cabin          327
Embarked         0
dtype: int64

<h6>Here, We have 86 null values in Age and 327 null values in cabin.</h6>

In [20]:
#df_test.head(20)
mean_value = df_test['Age'].mean()
df_test['Age'].fillna(value = mean_value, inplace=True)
df_test.head(20)

PassengerId  Pclass                                               Name  \
0           892       3                                   Kelly, Mr. James   
1           893       3                   Wilkes, Mrs. James (Ellen Needs)   
2           894       2                          Myles, Mr. Thomas Francis   
3           895       3                                   Wirz, Mr. Albert   
4           896       3       Hirvonen, Mrs. Alexander (Helga E Lindqvist)   
5           897       3                         Svensson, Mr. Johan Cervin   
6           898       3                               Connolly, Miss. Kate   
7           899       2                       Caldwell, Mr. Albert Francis   
8           900       3          Abrahim, Mrs. Joseph (Sophie Halaut Easu)   
9           901       3                            Davies, Mr. John Samuel   
10          902       3                                   Ilieff, Mr. Ylio   
11          903       1                         Jones, Mr. Charles Cresson   
12          904       1      Snyder, Mrs. John Pillsbury (Nelle Stevenson)   
13          905       2                               Howard, Mr. Benjamin   
14          906       1  Chaffee, Mrs. Herbert Fuller (Carrie Constance...   
15          907       2      del Carlo, Mrs. Sebastiano (Argenia Genovesi)   
16          908       2                                  Keane, Mr. Daniel   
17          909       3                                  Assaf, Mr. Gerios   
18          910       3                       Ilmakangas, Miss. Ida Livija   
19          911       3              Assaf Khalil, Mrs. Mariana (Miriam")"   

       Sex       Age  SibSp  Parch            Ticket     Fare Cabin Embarked  
0     male  34.50000      0      0            330911   7.8292   NaN        Q  
1   female  47.00000      1      0            363272   7.0000   NaN        S  
2     male  62.00000      0      0            240276   9.6875   NaN        Q  
3     male  27.00000      0      0            315154   8.6625   NaN        S  
4   female  22.00000      1      1           3101298  12.2875   NaN        S  
5     male  14.00000      0      0              7538   9.2250   NaN        S  
6   female  30.00000      0      0            330972   7.6292   NaN        Q  
7     male  26.00000      1      1            248738  29.0000   NaN        S  
8   female  18.00000      0      0              2657   7.2292   NaN        C  
9     male  21.00000      2      0         A/4 48871  24.1500   NaN        S  
10    male  30.27259      0      0            349220   7.8958   NaN        S  
11    male  46.00000      0      0               694  26.0000   NaN        S  
12  female  23.00000      1      0             21228  82.2667   B45        S  
13    male  63.00000      1      0             24065  26.0000   NaN        S  
14  female  47.00000      1      0       W.E.P. 5734  61.1750   E31        S  
15  female  24.00000      1      0     SC/PARIS 2167  27.7208   NaN        C  
16    male  35.00000      0      0            233734  12.3500   NaN        Q  
17    male  21.00000      0      0              2692   7.2250   NaN        C  
18  female  27.00000      1      0  STON/O2. 3101270   7.9250   NaN        S  
19  female  45.00000      0      0              2696   7.2250   NaN        C

<h6>Here, you can see the NaN has been replaced by mean value. See column no. 10.</h6>